subprojects/reporting/src/main/java/org/gradle/api/reporting/internal/DefaultReportContainer.java

```diff
@@ -51,7 +48,12 @@ public boolean isSatisfiedBy(T element) {
             }
         });
 
-        beforeChange(IMMUTABLE_VIOLATION_EXCEPTION);
+        whenObjectRemoved(IMMUTABLE_VIOLATION_EXCEPTION);
```

Comment:
Is this still necessary given the overloading of `assertMutableCollectionContents` below?